dso-l2/src/main/java/com/tc/server/TCServerImpl.java

```diff
@@ -137,6 +137,7 @@ public void stopIfActive(StopAction...restartMode) throws PlatformStopException
 
   @Override
   public void stop(StopAction...restartMode) {
+    GuardianContext.validate(Guardian.Op.STATE_CHANGE, "Stop invoked");
```

Comment:
Put the `validate` call in the right place, so that no matter where it's called from, we guard the operation